### DISCUSSION TITLE
npm install -g @angular/cli@11.0.4 is giving me errors! please help

### DISCUSSION THREAD

**Discussion by Mohlomi-M** [11ty/eleventy#2528] [Q&A]
npm WARN deprecated ini@1.3.5: Please update to ini >=1.3.6 to avoid a prototype pollution issue
npm WARN deprecated debug@4.2.0: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm ERR! code EEXIST
npm ERR! path C:\Users\Mohlomi\AppData\Roaming\npm\ng.ps1
npm ERR! Refusing to delete C:\Users\Mohlomi\AppData\Roaming\npm\ng.ps1: node_modules/@angular/cli/bin/ng symlink target is not controlled by npm C:\Users\Mohlomi\AppData\Roaming\npm\node_modules\@angular\cli
npm ERR! File exists: C:\Users\Mohlomi\AppData\Roaming\npm\ng.ps1
npm ERR! Remove the existing file and try again, or run npm
npm ERR! with --force to overwrite files recklessly.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Mohlomi\AppData\Roaming\npm-cache\_logs\2022-08-10T14_33_31_875Z-debug.log
PS C:\MissaSource\Src\Apps\missa> 
 *  History restored 

PS C:\MissaSource\Src\Apps\missa> npm install -g @angular/cli@11.0.4
npm WARN deprecated ini@1.3.5: Please update to ini >=1.3.6 to avoid a prototype pollution issue
npm WARN deprecated debug@4.2.0: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm ERR! code EEXIST
npm ERR! path C:\Users\Mohlomi\AppData\Roaming\npm\ng.ps1
npm ERR! Refusing to delete C:\Users\Mohlomi\AppData\Roaming\npm\ng.ps1: node_modules/@angular/cli/bin/ng symlink target is not controlled by npm C:\Users\Mohlomi\AppData\Roaming\npm\node_modules\@angular\cli
npm ERR! File exists: C:\Users\Mohlomi\AppData\Roaming\npm\ng.ps1
npm ERR! Remove the existing file and try again, or run npm
npm ERR! with --force to overwrite files recklessly.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Mohlomi\AppData\Roaming\npm-cache\_logs\2022-08-10T14_50_39_626Z-debug.log
PS C:\MissaSource\Src\Apps\missa>

**Comment by zachleat** [comment]
Is this related to Eleventy somehow? You might be on the wrong repo

**Comment by pdehaan** [comment]
@Mohlomi-M It looks like you're trying to globally install `@angular/cli` globally, but it's possibly already installed.

```sh
npm ERR! code EEXIST
npm ERR! path C:\Users\Mohlomi\AppData\Roaming\npm\ng.ps1
npm ERR! Refusing to delete C:\Users\Mohlomi\AppData\Roaming\npm\ng.ps1: node_modules/@angular/cli/bin/ng symlink target is not controlled by npm C:\Users\Mohlomi\AppData\Roaming\npm\node_modules@angular\cli
npm ERR! File exists: C:\Users\Mohlomi\AppData\Roaming\npm\ng.ps1
npm ERR! Remove the existing file and try again, or run npm
npm ERR! with --force to overwrite files recklessly.
```

Maybe try something in https://github.com/npm/npm/issues/13458
Otherwise, you might want to try asking in the https://github.com/angular/angular-cli project.
